entity-framework/core/providers/sqlite/value-generation.md

```diff
@@ -0,0 +1,71 @@
+---
+title: SQLite Database Provider - Value Generation - EF Core
+description: Value Generation Patterns Specific to the SQLite Entity Framework Core Database Provider
+author: AndriySvyryd
+ms.date: 09/26/2025
+uid: core/providers/sqlite/value-generation
+---
+# SQLite Value Generation
+
+This page details value generation configuration and patterns that are specific to the SQLite provider. It's recommended to first read [the general page on value generation](xref:core/modeling/generated-properties).
+
+## AUTOINCREMENT columns
+
+By convention, numeric primary key columns that are configured to have their values generated on add are set up with SQLite's AUTOINCREMENT feature. Starting with EF Core 10, SQLite AUTOINCREMENT is a first-class feature with full support through conventions and the Fluent API.
+
+### Configuring AUTOINCREMENT
+
+By convention, integer primary keys are automatically configured with AUTOINCREMENT when they are not composite and don't have a foreign key on them. However, you may need to explicitly configure a property to use SQLite AUTOINCREMENT when the property has a value conversion from a non-integer type, or when overriding conventions:
+
+[!code-csharp[Main](../../../../samples/core/Sqlite/ValueGeneration/SqliteAutoincrementWithValueConverter.cs?name=SqliteAutoincrementWithValueConverter&highlight=6)]
+
+## Disabling AUTOINCREMENT for default SQLite value generation
+
+In some cases, you may want to disable AUTOINCREMENT and use SQLite's default value generation behavior instead. You can do this using the Metadata API:
```

Comment:
Maybe just add a few words on when/why someone would want this (or at least link to the docs)? Otherwise that's just documenting our APIs without their meaning/why they exist.